### DISCUSSION TITLE
move GLib and GObject inside Vips

### DISCUSSION THREAD

**Discussion by jcupitt** [libvips/ruby-vips#268] [Ideas]
It would be neater to have the GLib and GObject modules inside Vips, see https://github.com/jcupitt/ruby-vips/issues/119#issuecomment-325286577

Though I wonder if this would be useful in practice. You can't have two glib or gobject libraries loaded into your program at once, since they both have plenty of global state, so two GLib modules would probably be a disaster.

**Comment by mvz** [comment]
I suppose this depends on whether you want ruby-vips to be loaded together with ruby-gnome2. IMHO, this should definitely be supported.

**Comment by jcupitt** [comment]
You're right, it should be possible. I'll have another go.

**Comment by jcupitt** [comment]
Aside: @mvz, I copied the ruby-vips inheritance stuff from your example (you probably noticed), so thank you very much!

https://github.com/jcupitt/ruby-vips/blob/master/example/inheritance_with_refcount.rb
